lib/bundle/brew_installer.rb

```diff
@@ -9,11 +9,18 @@ def initialize(name, options = {})
         @full_name = name
         @name = name.split("/").last
         @args = options.fetch(:args, []).map { |arg| "--#{arg}" }
+        @restart = options.fetch(:restart_service, false)
       else
         raise "Unable to install #{name}. Homebrew is not currently installed on your system"
       end
     end
 
+    def run
+      install_or_upgrade && restart
```

Comment:
How about `sudo` option, I think we need to change to DSL the support that. And in case the DSL change like i suggest, i don't think add optional params into  `install_or_upgrade` is the right way to do it